substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/Target_sun_security_ssl_TrustStoreManager.java

```diff
@@ -85,36 +89,79 @@ public void afterRegistration(AfterRegistrationAccess access) {
     final Set<X509Certificate> trustedCerts;
     final KeyStore trustedKeyStore;
 
+    private static volatile Boolean truststoreSysPropSet;
+
     TrustStoreManagerSupport(Set<X509Certificate> trustedCerts, KeyStore trustedKeyStore) {
         this.trustedCerts = trustedCerts;
         this.trustedKeyStore = trustedKeyStore;
     }
+
+    static boolean useEmbeddedCerts() {
+        if (truststoreSysPropSet != null) {
```

Comment:
I don't think you can do this that easily:
* The `trustStore` system property can be changed (set and unset) at any time at run time. You need to always reflect the current correct value.
* Separating the check "is system property set" from the actual usage of the property in the JDK code has the problem of race conditions: if the system property is changed in between, the wrong value is used.
So I don't think you can have a proper implementation without deep integration into `TrustStoreManager` and `TrustStoreDescriptor` of the JDK.